### PR TITLE
fix: use legacy syntax for decorators

### PR DESCRIPTION
**Pull request by anehx**
This commit rewrites the decorators `withParachute` and `queryParam` to
use the stage 1 legacy decorator syntax since ember uses that syntax for
now. For more infos see https://github.com/ember-decorators/ember-decorators/blob/master/README.md#specs.

This will fix the usage of `ember-parachute`s decorators with newer versions of ember / ember-decorators.

### CODE REVIEW

**Comment by anehx** [comment]
This also drops support for node 6 since it's EOL is 2019-04-30 and `ember-decorators` v6+ doesn't support it

**Comment by anehx** [comment]
@offirgolan I don't really see the reason for the failing tests.. Maybe cleaning the Travis CI cache would help?

**Comment by anehx** [comment]
Sadly I just learned that this doesn't really work the way I intended it.. The routes [`setup`](https://github.com/emberjs/ember.js/blob/v3.14.1/packages/%40ember/-internals/routing/lib/system/route.ts#L898) hook which initializes the QPs values uses a computed property [`_qp`](https://github.com/emberjs/ember.js/blob/v3.14.1/packages/%40ember/-internals/routing/lib/system/route.ts#L2266) which uses the `owner.lookup` method to get the controller definition. If the controllers `queryParams` property is defined via `Controller.reopen` it does not exist then (not quite sure why, since controllers are supposed to be a singleton).

Long story short, this won't work properly. I don't have any more ideas how to solve it. If someone has an idea I'm very willing to try it out..
